pybamm/plotting/quick_plot.py

```diff
@@ -154,6 +154,10 @@ def __init__(
                     f"No default output variables provided for {models[0].name}"
                 )
 
+        # check variables have been provided after any serialisation
+        if any(len(m.variables) == 0 for m in models):
+            raise AttributeError("Variables not provided by the serialised model")
```

Comment:
could just be a model without any variables? Perhaps change error to "No variables to plot"